data/pigui/libs/icons.lua

```diff
@@ -14,6 +12,20 @@ local function get_wide_icon_tex_coords(icon)
 	return Vector2(rem / count, quot/count), Vector2((rem+2) / count, (quot+1)/count)
 end
 
+local icons_texture_small = pigui:LoadTextureFromSVG(pigui.DataDirPath({"icons", "icons.svg"}), 16 * 24, 16 * 24)
+local icons_texture_med = pigui:LoadTextureFromSVG(pigui.DataDirPath({"icons", "icons.svg"}), 16 * 32, 16 * 32)
+local icons_texture_large = pigui:LoadTextureFromSVG(pigui.DataDirPath({"icons", "icons.svg"}), 16 * 64, 16 * 64)
+
+function ui.get_icons_texture(size)
```

Comment:
Let say, size = 22. Why scaling from 24 pixels is better than from 64?

Reply:
Moire patterns! More specifically, these icons are not created with mipmaps, so scaling from 64px to 22px causes a much more jagged and incorrect icon than scaling from 24px to 22px. You can see this with the navpoint icon and certain icons in the worldview - they retain much more of their shape when rasterized at 24px instead of at 64px and having the rasterized icon scaled down to <16x16.

Reply:
It's definitely not perfect; some of our icons were never intended to display at less than 64px, but for text icons and other buttons it makes a small but noticeable difference. The first instinct is to think that the icon is more blurry when rendered from a smaller buffer, but when you go back the sharp, jagged edges and downsampling artifacts are *very* noticeable.